packages/next/src/utils.ts

```diff
@@ -1242,30 +1242,36 @@ async function getSourceFilePathFromPage({
   // value used during next build
   const extensionsToTry = pageExtensions || ['js', 'jsx', 'ts', 'tsx'];
 
-  let fsPath = path.join(workPath, 'pages', page);
-  if (await usesSrcDirectory(workPath)) {
-    fsPath = path.join(workPath, 'src', 'pages', page);
-  }
-
-  if (fs.existsSync(fsPath)) {
-    return path.relative(workPath, fsPath);
-  }
-  const extensionless = fsPath.slice(0, -3); // remove ".js"
+  for (const pageType of ['pages', 'app']) {
+    let fsPath = path.join(workPath, pageType, page);
+    if (await usesSrcDirectory(workPath)) {
+      fsPath = path.join(workPath, 'src', pageType, page);
+    }
 
-  for (const ext of extensionsToTry) {
-    fsPath = `${extensionless}.${ext}`;
     if (fs.existsSync(fsPath)) {
       return path.relative(workPath, fsPath);
     }
-  }
+    const extensionless = fsPath.slice(0, -3); // remove ".js"
```

Comment:
`slice()` scares me. I know you just moved this line, but I'd feel more comfortable if this replacement was more explicit:

```suggestion
    const extensionless = fsPath.replace(path.extname(fsPath), '');
```

Reply:
100%! Thanks for catching this!